test/eip721/eip721.js

```diff
@@ -0,0 +1,605 @@
+const { assertRevert } = require('../helpers/assertRevert');
+
+const testEip721 = artifacts.require('TestEIP721Implementation');
+const testReceiver = artifacts.require('TestReceiver');
+const TestNonStandardReceiver = artifacts.require('TestNonStandardReceiver');
+let nft;
+let admin;
+
+// NOTE: This disable is for all the event logs args having underscores
+/* eslint-disable no-underscore-dangle */
+
+contract('TestEIP72Implementation', (accounts) => {
+  beforeEach(async () => {
+    nft = await testEip721.new({ gas: 6720000, from: accounts[0] });
+    admin = await nft.admin.call();
+  });
+
+  it('creation: create one token', async () => {
+    const result = await nft.createToken(admin, { from: accounts[0] });
+
+    // verify Transfer event (from == 0 if creating token)
+    assert.strictEqual(result.logs[0].event, 'Transfer');
+    assert.strictEqual(result.logs[0].args._from, '0x0000000000000000000000000000000000000000');
+    assert.strictEqual(result.logs[0].args._to, admin);
+    assert.strictEqual(result.logs[0].args._tokenId.toString(), '0');
+
+    const totalSupply = await nft.totalSupply.call();
+    const adminBalance = await nft.balanceOf.call(admin);
+    const owner = await nft.ownerOf.call(0);
+
+    assert.strictEqual(totalSupply.toString(), '1');
+    assert.strictEqual(adminBalance.toString(), '1');
+    assert.strictEqual(admin, owner);
+  });
+
+  it('creation: create one token then retrieve one that does not exist (should fail)', async () => {
+    await nft.createToken(admin, { from: accounts[0] });
+
+    const owner = await nft.ownerOf.call(0);
+    assert.strictEqual(admin, owner);
+    await assertRevert(nft.ownerOf.call(1));
+  });
+
+  it('creation: create multiple tokens to one user', async () => {
+    await nft.createToken(accounts[0], { from: accounts[0] });
```

Comment:
looks like the `createToken` method signature only takes one, so maybe the second arg isn't needed?